src/components/Swap/Toolbar/index.tsx

```diff
@@ -72,9 +83,23 @@ export default memo(function Toolbar() {
       }
       if (trade?.inputAmount && trade.outputAmount) {
         return impact?.warning ? (
-          <Caption.PriceImpact impact={impact} />
+          <Caption.PriceImpact
+            impact={impact}
+            expanded={expanded}
+            toggleExpanded={() => {
+              setExpanded(!expanded)
```

Comment:
Do not ever pass an old value into a state setter, as it risks render loops.
Instead, use a setter function to retrieve the old value. In this case:

    setExpanded((expanded) => !expanded)
    
The reason to not set using the old value is that, in a hook, it could look like this, triggering rerenders because it depends on what it is setting: `useEffect(() => setExpanded(!expanded), [expanded])`.

Reply:
^ eh i disagree here, `expanded` is a boolean primitive value and there's no race condition with state setting here, so this feels unnecessary